pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -622,3 +622,21 @@ func (dr *DashboardServiceImpl) DeleteInFolder(ctx context.Context, orgID int64,
 }
 
 func (dr *DashboardServiceImpl) Kind() string { return entity.StandardKindDashboard }
+
+func (dr *DashboardServiceImpl) CleanUpDeletedDashboards(ctx context.Context) (int64, error) {
+	var deletedDashboardsCount int64
+	deletedDashboards, err := dr.dashboardStore.GetSoftDeletedDashboards(ctx)
+	if err != nil {
+		return 0, err
+	}
+	for _, dashboard := range deletedDashboards {
```

Comment:
shouldn't it be an atomic transaction?

Reply:
I'm not sure, I had that concern but why not delete what we can and keep only those that are failing? it's an automated process 🤔